src/main/java/meditracker/command/SaveCommand.java

```diff
@@ -46,33 +46,56 @@ public SaveCommand(String arguments)
     }
 
     /**
-     * Saves MediTracker information as JSON at its default save location specified in MediTrackerConfig.
+     * Saves MediTracker information to the default save location specified in `MediTrackerConfig`.
+     * Includes the Medication data (in JSON) and DailyMedication data (in txt).
      */
-    private void saveJsonToDefaultLocation() {
-        boolean isSaveSuccessful = FileReaderWriter.saveMediTrackerData(null);
+    private void saveToDefaultLocation() {
+        boolean isSaveSuccessful = FileReaderWriter.saveMedicationData(null);
         if (!isSaveSuccessful) {
-            System.out.println("An error occurred while saving. Data is not saved.");
+            System.out.println("An error occurred while saving Medication data. Data is not saved.");
+            return;
         } else {
             Path defaultJsonPath = MediTrackerConfig.getDefaultJsonSaveFilePath();
             Path absoluteJsonPath = defaultJsonPath.toAbsolutePath();
-            System.out.println("Data successfully saved to: " + absoluteJsonPath);
+            System.out.println("Medication Data successfully saved to: " + absoluteJsonPath);
             System.out.println("Use the -o flag to save the data to another location.");
         }
+
+        isSaveSuccessful = FileReaderWriter.saveDailyMedicationData(null);
+        if (!isSaveSuccessful) {
+            System.out.println("An error occurred while saving DailyMedication Data. Data is not saved.");
+        } else {
+            Path dailyMedPath = MediTrackerConfig.getDailymedFilePath(null);
+            Path absoluteDailyMedPath = dailyMedPath.toAbsolutePath();
+            System.out.println("DailyMedication Data successfully saved to: " + absoluteDailyMedPath);
+        }
     }
 
     /**
-     * Saves MediTracker information as JSON at the specified path.
+     * Saves MediTracker information to the specified location.
+     * Includes the Medication data (in JSON) and DailyMedication data (in txt).
+     * The DailyMedication data will be saved relative to the JSON file.
      *
-     * @param saveFilePath The location to save to.
+     * @param jsonSaveFilePath The location of the JSON file to save to.
      */
-    private void saveJsonToSpecifiedLocation(Path saveFilePath) {
-        boolean isSaveSuccessful = FileReaderWriter.saveMediTrackerData(saveFilePath);
+    private void saveToSpecifiedLocation(Path jsonSaveFilePath) {
+        boolean isSaveSuccessful = FileReaderWriter.saveMedicationData(jsonSaveFilePath);
         if (!isSaveSuccessful) {
-            System.out.println("An error occurred while saving. Data is not saved.");
+            System.out.println("An error occurred while saving Medication Data. Data is not saved.");
+            return;
```

Comment:
Same here. Since both saving process should remain independent (i.e. should attempt saveDailyMedicationData), even if saveMedicationData fails, code should not return here.

Reply:
resolved: e8ab6759f4e39b486aa4286098edb631b93bc509